### PR TITLE
Extract StorybookArgs type to separate file and use it in other stories too

### PR DESCRIPTION
**Pull request by PavelVanecek**
followup from https://github.com/recharts/recharts/pull/3759#discussion_r1327816026

Adding the type helps to find out what the properties are and what shape is expected.

Storybook website has it documented, and provides typescript types, but for some reason that type is not exported in code.

## Description

Add type to storybooks

## Related Issue

https://github.com/recharts/recharts/pull/3759

## Motivation and Context

Helps to write storybooks.

## How Has This Been Tested?

lint, ts

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: storybook/StorybookArgs.ts
Comment (nikolasrieble) [inline storybook/StorybookArgs.ts:10]: Just wondering: Would you like to do a PR in storybook to export those types?

File: storybook/StorybookArgs.ts
Comment (nikolasrieble) [inline storybook/StorybookArgs.ts:6]: ❤️ Love this. Thank you.

**Review by nikolasrieble — APPROVED** [review]
Thank you a lot, this is great! Certainly makes all our life easier.

File: storybook/StorybookArgs.ts
Comment (PavelVanecek) [reply storybook/StorybookArgs.ts:10]: Yeah I'll look into that. I am afraid there's some roadblock - why else would the storybook people add the type in the docs but not in the code? But it's worth a try at least.

**Comment by PavelVanecek** [comment]
Thanks for the review @nikolasrieble. I wonder what's the usual workflow? I don't think I have merge permissions myself. Are we waiting for two approvers?

**Comment by ckifer** [comment]
Generally @nikolasrieble or myself will review and merge, sometimes waiting for more reviews depending on the size/implications of a change. Eventually I hope to have more maintainers will merge permissions

**Review by ckifer — APPROVED** [review]
Very nice, thanks for doing this

**Pull request merged**
